### PR TITLE
[dashboards] Add resources for log, apm and process query in legacy dashboards

### PR DESCRIPTION
**Pull request by jbenais**
- Fixes issue: https://github.com/terraform-providers/terraform-provider-datadog/issues/120
- Adds new tests for the new log, apm and process query params.
- Updates the documentation for dashboards, screenboards and timeboards resources.

*This functionality relies on a change in the Go client that is in a merged PR: zorkian/go-datadog-api#251*

### CODE REVIEW

File: datadog/resource_datadog_screenboard.go [hunk 1]
```diff
@@ -88,6 +88,115 @@ func resourceDatadogScreenboard() *schema.Resource {
...
+							"aggregation": {
+								Type:     schema.TypeString,
+								Required: true,
+							},
+							"facet": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"interval": {
+								Type:     schema.TypeInt,
```
Comment (enbashi) [inline datadog/resource_datadog_screenboard.go]: This looks wrong for the legacy API. I think it should be a string. Try to import a dashboard created by the UI. You will get:
`Error: datadog_screenboard.import (import id:xxx-xxx-xxx): import datadog_screenboard.import (id: xxx-xxx-xxx): json: cannot unmarshal string into Go struct field TileDefApmOrLogQueryCompute.interval of type int

**Comment by nmuesch** [comment]
Hey @jbenais thanks for the PR! The golang client was released + bumped in this provider. Can you rebase from master and we can work on getting this reviewed?

File: datadog/resource_datadog_screenboard.go [hunk 2]
```diff
@@ -741,6 +853,96 @@ func buildTileDefRequestsConditionalFormats(source interface{}) []datadog.Condit
...
+	// Index
+	d := datadog.TileDefApmOrLogQuery{
+		Index: datadog.String(datadogQuery["index"].(string)),
+	}
+
+	// Compute
+	terraformCompute := datadogQuery["compute"].(map[string]interface{})
+	datadogCompute := datadog.TileDefApmOrLogQueryCompute{}
+	if aggr, ok := terraformCompute["aggregation"].(string); ok && len(aggr) != 0 {
+		datadogCompute.Aggregation = datadog.String(aggr)
```
Comment (nmuesch) [inline datadog/resource_datadog_screenboard.go]: Should we use the SetX methods from the datadog accessors for these?

Comment (nmuesch) [reply datadog/resource_datadog_screenboard.go]: Not sure the history, but I notice we aren't really using any of the GetX or SetX methods in these resources.

File: datadog/resource_datadog_timeboard.go [hunk 3]
```diff
@@ -13,15 +13,127 @@ import (
 )
 
 func resourceDatadogTimeboard() *schema.Resource {
+	apmOrLogQuery := &schema.Schema{
```
Comment (nmuesch) [inline datadog/resource_datadog_timeboard.go]: Is this the same schema definition thats also in screenboards? If so do you think its worth it to find a way to refactor this so both reference the same definition?

File: datadog/resource_datadog_timeboard_test.go [hunk 4]
```diff
@@ -17,25 +17,26 @@ resource "datadog_timeboard" "acceptance_test" {
   description = "Created using the Datadog provider in Terraform"
   read_only = true
   graph {
-    title = "Top System CPU by Docker container"
```
Comment (nmuesch) [inline datadog/resource_datadog_timeboard_test.go]: Looks like the indentation here got updated with tabs vs spaces.

File: datadog/resource_datadog_timeboard.go (hunk 3 above)
Comment (jbenais) [reply datadog/resource_datadog_timeboard.go]: It's _almost_ the same, but some fields like `compute->interval` (that is an integer in timeboards, and a string in screenboards) are different.

File: website/docs/r/screenboard.html.markdown [hunk 5]
```diff
@@ -592,6 +657,34 @@ The nested `style` blocks has the following structure:
 - `width` - (Optional) Line width. Possible values: "thin", "normal", "thick". Default: "normal".
 - `type` - (Optional) Type of line drawn. Possible values: "dashed", "solid", "dotted". Default: "solid".
 
+### Nested `widget` `tile_def` `request` `apm_query` and `log_query` blocks
+
+Nested `apm_query` and `log_query` blocks have the following structure (Visit the [ Graph Primer](https://docs.datadoghq.com/graphing/) for more information about these values):
+
+- `index` - (Required)
```
[the remark this answers is not in the document]
Comment (jbenais) [reply website/docs/r/screenboard.html.markdown]: I think it's already done above (i.e. lign 602 to 604)

File: datadog/resource_datadog_timeboard.go [hunk 6]
```diff
@@ -371,15 +486,111 @@ func buildTemplateVariables(terraformTemplateVariables *[]interface{}) *[]datado
...
+	if v, ok := terraformQuery["metric"].(string); ok && len(v) != 0 {
+		datadogQuery.SetMetric(v)
+	}
+	if v, ok := terraformQuery["search_by"].(string); ok {
+		datadogQuery.SetSearchBy(v)
+	}
+
+	if terraformFilterBys, ok := terraformQuery["filter_by"].([]interface{}); ok {
+		datadogFilterbys := make([]string, len(terraformFilterBys))
+		for i, filtrBy := range terraformFilterBys {
```
Comment (nmuesch) [inline datadog/resource_datadog_timeboard.go]: filter?

**Pull request merged**